packages/rol/adapters/thyra/src/function/ROL_ThyraProductME_Constraint_SimOpt.hpp

```diff
@@ -331,14 +397,15 @@ class ThyraProductME_Constraint_SimOpt : public Constraint_SimOpt<Real> {
     }
     const Teuchos::RCP<Thyra::LinearOpWithSolveBase<Real> > jacobian = lows_factory->createOp();
 
-    if(updateJacobian1)
+    if(computeJacobian1)
     {
       outArgs.set_W_op(lop);
       thyra_model.evalModel(inArgs, outArgs);
       outArgs.set_W_op(Teuchos::null);
       inArgs.set_x(Teuchos::null);
       jac1 = lop;
-      updateJacobian1 = false;
+
+      computeJacobian1 = false;
```

Comment:
The use of computeJacobian1 looks like it is introducing "state" into the model evaluator. Is that the case? It is something we have tried to avoid in the design.

Reply:
Unfortunately yes. I don't see an easy way to overcome this. Alternatively I could write this info in the parameter list provided by the application.. but it seems even less clean to me.
Note that if we used something like an observer object provided by, e.g., Piro, then we would need ROL to depend on Piro.. which is bad. The alternative is to move the Thyra adapters into Piro. Which could be a better solution. 
I know that ROL is going to refactor the way the update function works. Not sure that will help (@dridzal ).